tools/nixpkgs-hammer

```diff
@@ -23,10 +23,16 @@ def nix_eval(expr: str) -> str:
 
 
 def nix_eval_json(expr: str, as_string=False):
-    result = subprocess.check_output(
-        [NIX_INSTANTIATE_PATH, '--strict', '--json', '--eval', '-E', expr],
-        encoding='utf-8',
-    )
+    try:
+        result = subprocess.check_output(
+            [NIX_INSTANTIATE_PATH, '--strict', '--json', '--eval', '--show-trace', '-E', expr],
+            encoding='utf-8',
+        )
+    except subprocess.CalledProcessError as e:
+        print(red(subprocess.list2cmdline(e.cmd)))
+        print(f"Returned nonzero exit status {e.returncode}")
+        exit(1)
```

Comment:
Maybe there's a better way to show this to the user so that we can distinguish crashes/bugs from expected errors because of user typos like:

```
$  nix run -c nixpkgs-hammer -f ~/projects/nixpkgs something.that.doesnt.exist
error: attribute 'something' missing, at (string):3:11
/nix/store/859dm0ls580aknphgib62xqwjavimj8j-nix-2.3.10/bin/nix-instantiate --eval -E "
let
    drv = (import /home/mcgibbon/projects/nixpkgs { }).something.that.doesnt.exist;
    pname = builtins.unsafeGetAttrPos \"pname\" drv;
in
    if pname != null then
        pname
    else
        builtins.unsafeGetAttrPos \"name\" drv
"
Returned nonzero exit status 1
```

Reply:
Hmm, must be a regression. It is supposed to be caught by https://github.com/jtojnar/nixpkgs-hammering/blob/6f2c7e289256983c8da70feea15d6c2d696551f8/tools/nixpkgs-hammer#L104-L116